quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -245,6 +248,22 @@ impl SourceParams {
     pub fn void() -> Self {
         Self::Void(VoidSourceParams)
     }
+
+    pub fn use_shard_api(&self) -> bool {
```

Comment:
This need some documentation explaining what this is about.

Reply:
I think we could use a more explicit name and semantic. What do you think about replacing this with:

```
pub fn checkpoint_store() -> CheckpointStore
```

where 

```
pub enum CheckpointStore {
    ShardApi,
    IndexMetadata,
}
```